pkg/api/helpers.go

```diff
@@ -38,8 +55,12 @@ var Semantic = conversion.EqualitiesOrDie(
 		if b.Amount == nil && a.MilliValue() == 0 {
 			return true
 		}
+		if a.Amount == nil || b.Amount == nil {
```

Comment:
It's subtle-- if both are nil, the first if will catch it; MilliValue() is safe to call and returns zero in that case.